ksqldb-common/src/main/java/io/confluent/ksql/util/ConsistencyOffsetVector.java

```diff
@@ -165,12 +225,34 @@ public int hashCode() {
 
   @Override
   public String toString() {
-    return "OffsetVectorImpl{"
-      + "version=" + version
-      + ", offsetVector=" + offsetVector
-      + ", rwLock=" + rwLock
-      + '}';
+    final StringBuilder sb = new StringBuilder("ConsistencyOffsetVector{");
+    sb.append("version=").append(version);
+    sb.append(", offsetVector=").append(offsetVector);
+    sb.append(", rwLock=").append(rwLock);
+    sb.append('}');
+    return sb.toString();
+  }
+
+  public static boolean isConsistencyVectorEnabled(
+      final KsqlConfig ksqlConfig,
+      final Map<String, Object> overrides
+  ) {
+    if (overrides.containsKey(KsqlConfig.KSQL_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR_ENABLED)) {
+      return Boolean.TRUE.equals(overrides.get(
+          KsqlConfig.KSQL_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR_ENABLED));
+    } else {
+      return ksqlConfig.getBoolean(KsqlConfig.KSQL_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR_ENABLED);
+    }
   }
-}
 
+  public static boolean isConsistencyVectorEnabled(final Map<String, Object> requestProperties) {
+    final Object consistencyEnabled
```

Comment:
Normally request properties are defined in `KsqlRequestConfig`.  We don't usually send KsqlConfigs up as a request property, but rather a normal property.

Reply:
I wanted to avoid creating two different configs for the same thing so reused the server config in the client as well. I can change it if it is too ugly not the right way to do it